tools/buildmgr/cbuildgen/config/IAR.8.50.6.cmake

```diff
@@ -0,0 +1,157 @@
+# Version: 0.10.4
+# Date: 2022-01-21
+# This file maps the CMSIS project options to toolchain settings.
+#
+#   - Applies to toolchain: IAR ARM C/C++ Compiler V8.50.6
+
+############### EDIT BELOW ###############
+# Set base directory of toolchain
+set(TOOLCHAIN_ROOT)
+set(EXT)
+
+############ DO NOT EDIT BELOW ###########
+
+set(AS ${TOOLCHAIN_ROOT}/iasmarm${EXT})
+set(CC ${TOOLCHAIN_ROOT}/iccarm${EXT})
+set(CXX ${TOOLCHAIN_ROOT}/iccarm${EXT})
+set(LD ${TOOLCHAIN_ROOT}/ilinkarm${EXT})
+set(AR ${TOOLCHAIN_ROOT}/iarchive${EXT})
+set(OC ${TOOLCHAIN_ROOT}/ielftool${EXT})
+
+# Core Options
+
+if(CPU STREQUAL "Cortex-M0")
+  set(IAR_CPU "Cortex-M0")
+elseif(CPU STREQUAL "Cortex-M0+")
+  set(IAR_CPU "Cortex-M0+")
+elseif(CPU STREQUAL "Cortex-M1")
+  set(IAR_CPU "Cortex-M1")
+elseif(CPU STREQUAL "Cortex-M3")
+  set(IAR_CPU "Cortex-M3")
+elseif(CPU STREQUAL "Cortex-M4")
+  set(IAR_CPU "Cortex-M4")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "Cortex-M4.fp.sp")
+  endif()
+elseif(CPU STREQUAL "Cortex-M7")
+  set(IAR_CPU "Cortex-M7")
+  if(FPU STREQUAL "DP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp.dp")
+  elseif(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp.sp")
+  endif()
+elseif(CPU STREQUAL "Cortex-M23")
+  set(IAR_CPU "Cortex-M23")
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+elseif(CPU STREQUAL "Cortex-M33")
+  set(IAR_CPU "Cortex-M33")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp")
+  endif()
+  if(DSP STREQUAL "NO_DSP")
+    set(IAR_CPU "${IAR_CPU}.no_dsp")
+  endif()
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+elseif(CPU STREQUAL "Cortex-M35P")
+  set(IAR_CPU "Cortex-M35P")
+  if(FPU STREQUAL "SP_FPU")
+    set(IAR_CPU "${IAR_CPU}.fp")
+  endif()
+  if(DSP STREQUAL "NO_DSP")
+    set(IAR_CPU "${IAR_CPU}.no_dsp")
+  endif()
+  if(TZ STREQUAL "NO_TZ")
+    set(IAR_CPU "${IAR_CPU}.no_se")
+  endif()
+endif()
+if(NOT DEFINED IAR_CPU)
+  message(FATAL_ERROR "Error: CPU is not supported!")
+endif()
+
+# Assembler
+
+set(ASM_CPU "--cpu ${IAR_CPU}")
+set(ASM_FLAGS)
+
+foreach(ENTRY ${DEFINES})
+  string(APPEND ASM_DEFINES "-D${ENTRY}=1 ")
+endforeach()
```

Comment:
Have you considered that a `define` may have a `key=value` pair?
And that `value` may require to preserve quotation marks up to the toolchain command line?

For example:
```
.cprj:
<defines>MBEDTLS_CONFIG_FILE=\"aws_mbedtls_config.h\"</defines>

CMakeLists.txt:
set(DEFINES
  MBEDTLS_CONFIG_FILE=\"aws_mbedtls_config.h\"
)

build.ninja:
-DMBEDTLS_CONFIG_FILE=\"aws_mbedtls_config.h\"
```

Please check if this would work with the IAR compiler.